src/Storages/StorageURL.cpp

```diff
@@ -271,6 +206,72 @@ namespace
             }
         }
 
+        static std::unique_ptr<ReadBuffer> getFirstAvailableURLReadBuffer(
+            const std::vector<String> & urls,
+            ContextPtr context,
+            const URIParams & params,
+            const String & http_method,
+            std::function<void(std::ostream &)> callback,
+            const ConnectionTimeouts & timeouts,
+            const String & compression_method,
+            const ReadWriteBufferFromHTTP::HTTPHeaderEntries & headers,
+            bool skip_url_not_found_error)
+        {
+            Poco::Net::HTTPBasicCredentials credentials{};
```

Comment:
In ReadWriteBufferFromHTTP we will save the reference to `credentials` and this reference will be invalid after returning from `getFirstAvailableURLReadBuffer`